[valhrd] iP

valhrd opened the pull request:
# Project Yapper 🗣️🗣️🗣️

> "10/10 yapping session"

Yapper is designed to help you stay organized and efficient, all while reducing the mental load of repetitive tasks.

### Why Yapper?
- **Quick Setup**: Get started in minutes.
- **User-Friendly**: Designed with simplicity in mind.
- **Performance**: Optimized for speed and reliability.

### Steps to Get Started
1. Download the JAR file from [here](https://github.com/valhrd/ip/releases/tag/A-Jar).
2. Open the terminal and go to directory where the file is located.
3. Execute `java -jar Yapper.jar`.
4. Start using!

### Key Features
- [ ] Task management
- [ ] Event scheduling
- [ ] Convenient task deletion
- [ ] Real time saving

If you want to see the source code just click [here](https://github.com/valhrd/ip) and run from the Launcher class:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

valhrd committed:
Renamed file and provided skeletal code

valhrd committed:
Level-1: Echo

valhrd committed:
Level-2: Add, list

valhrd committed:
Level-3: Mark as Done

valhrd committed:
Level-4: ToDos, Events, Deadlines

valhrd committed:
A-TextUiTesting

valhrd committed:
Level-5: Handle Errors

valhrd committed:
Level-6: Delete

valhrd committed:
A-Enums: Use Enums

valhrd committed:
Add save feature

valhrd committed:
Change method name

valhrd committed:
Update writeToFile method

valhrd committed:
Add formatting dates and times

valhrd committed:
Encapsulate methods

valhrd committed:
Add classes

valhrd committed:
Add classes

valhrd committed:
Package exceptions and main

valhrd committed:
Merge branch 'add-gradle-support'

valhrd committed:
Add gradle and unit tests

valhrd committed:
Add dependencies for gradle

valhrd committed:
Add JavaDoc

valhrd committed:
Add more JavaDoc

valhrd committed:
Add find command and functionality

valhrd committed:
Merge branch 'branch-Level-9'

valhrd committed:
Update find command and functionality

valhrd committed:
Checkstyle existing classes

valhrd committed:
Checkstyle used

valhrd committed:
Add JavaFX GUI for chatbot

valhrd committed:
Add new icons and styling

valhrd committed:
Update bye functionality

valhrd committed:
Update bye functionality

valhrd committed:
Update features

valhrd committed:
no message

valhrd committed:
Edit description of findTasks method in TaskList class

valhrd committed:
Fix a bug for deadline and event commands

waihongteh commented on src/main/java/yapper/app/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+ * Represents a deadline task in the Yapper chatbot application.
+ */
+public class Deadline extends Task {
+    private String by;
```
It might be better to use a noun here? Maybe something like deadline.

valhrd replied:
Will change field name to "deadline" to make things clearer

waihongteh commented on src/main/java/yapper/app/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+     * @param by   the date and/or time by which the task must be completed
+     * @throws YapperException if the deadline is empty
+     */
+    public Deadline(String desc, String by) {
```
I personally will prefer description instead of desc as it is more intuitive, but desc looks good still!

valhrd replied:
Will change field name to "description"

waihongteh commented on src/main/java/yapper/app/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
db does not sound very understandable, maybe dialogBox would be better?

valhrd replied:
The scope of "db" does not exceed the few lines of code it's used in, so this should not be an issue, although in general more descriptive variable names are better.

If you'd like, you can also refer to the section regarding scratch variables in the coding standard:
https://se-education.org/guides/conventions/java/intermediate.html

waihongteh commented on src/main/java/yapper/app/MainWindow.java:
```diff
@@ -0,0 +1,63 @@
+    private Yapper yapper;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/you.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/icon.png"));
```
Did you forget to change this to yapperImage?

valhrd replied:
Yes, woops. Will change.

waihongteh commented on src/main/java/yapper/app/Task.java:
```diff
@@ -0,0 +1,126 @@
+            }
+        }
+
+        for (String fm : timeFormats) {
```
Maybe format is more preferrable than fm?

valhrd replied:
Will change, thanks.

waihongteh commented on src/main/java/yapper/app/TaskList.java:
```diff
@@ -0,0 +1,180 @@
+     *
+     * @return a string representing "task" or "tasks"
+     */
+    public String taskPlural() {
```
Since it is a method that converts task to plural maybe verb is better than noun here as the name

valhrd replied:
Will change to a more descriptive name.

waihongteh commented on src/main/java/yapper/app/Ui.java:
```diff
@@ -0,0 +1,67 @@
+    /**
+     * Displays the introduction message and prompts the user.
+     */
+    public static String intro() {
```
introduceMyself() might be better here as it is a method.

valhrd replied:
Yes, it should be more descriptive, thank you for the suggestion.

waihongteh reviewed (commented):
Goodjob overall! Naming looks mostly fine to me.

valhrd committed:
Change field and method names to be more descriptive

valhrd committed:
Change taskPlural to pluralise

valhrd committed:
Change taskPlural to pluralise

valhrd committed:
Add assertions to various methods

valhrd committed:
Refactor and add additional methods to increase readability

The code block responsible for parsing in a time to reformat it into date/time format had several for loops and try-catch blocks.

Breaking this method up and refactoring it into 4 methods improves the code quality and readability

As a step to improving code quality, let's try to extract for loops and other code fragments and refactor them into separate function calls. Doing so will make the code more readable and concise.

valhrd committed:
Change findTasks method to use Java streams to filter out tasks containing a given keyword

valhrd committed:
Merge pull request #1 from valhrd/branch-A-Assertions

Add assertions to various methods

valhrd committed:
Merge branch 'branch-A-CodeQuality' into master

valhrd committed:
Merge pull request #2 from valhrd/master

Code Quality Check

valhrd committed:
Merge branch 'branch-A-Streams'

valhrd committed:
Merge pull request #3 from valhrd/master

Streams

valhrd committed:
Merge pull request #4 from valhrd/branch-A-CodeQuality

Merge pull request #2 from valhrd/master

valhrd committed:
Merge pull request #5 from valhrd/branch-A-Streams

Branch a streams

valhrd committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/yapper/app/Storage.java
#	src/main/java/yapper/app/Task.java

valhrd committed:
Resolve merge conflicts

valhrd committed:
Add Command class and child classes to abstract chatbot commands.

The Yapper class handles a substantial amount of the command logic, including identifying deadlines, from and to dates and so on.

Creating a Command class and a CommandList class helps to condense the amount of code in the Yapper class and makes it more readable.

valhrd committed:
Abstract Yapper commands and add bind and unbind commands to help users add custom commands.

There are several commands for the Yapper chatbot and putting them all into one method with a big if-else block becomes messy and unreadable.

We should try to abstract and turn each command into it's own class and package them separately so code becomes more readable and easier to manage.

As a step towards making the code more readable, let's turn all the commands into classes so that binding and unbinding new commands can also be made easier.

valhrd committed:
Change UI background

valhrd committed:
Change error responses

valhrd committed:
Add command to clear the task list

valhrd committed:
Add UI screenshot for website

valhrd committed:
Update README.md

valhrd committed:
Rename Ui.png.png to Ui.png

valhrd committed:
Update README.md

valhrd committed:
Add Ui.png

valhrd committed:
Merge branch 'master' of https://github.com/valhrd/ip

valhrd committed:
Update user guide

valhrd committed:
Update user guide

valhrd committed:
Update user guide

valhrd committed:
Update user guide

valhrd committed:
Update user guide

valhrd committed:
Update user guide

valhrd committed:
Add yml file for continuous integration

valhrd committed:
Edit files to satisfy checkstyle

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Modify files for continuous integration

valhrd committed:
Update docs content

valhrd committed:
Update docs content

valhrd committed:
Update docs content

valhrd committed:
Add more unit testing

valhrd committed:
Update README

valhrd committed:
Update README

valhrd committed:
Update JavaDoc
